docs/go_api.md

```diff
@@ -0,0 +1,342 @@
+  let r = lcm(a, b);
+  return r;
+}
+
+#[wasm_bindgen]
+pub fn sha3_digest(v: Vec<u8>) -> Vec<u8> {
+  return Sha3_256::digest(&v).as_slice().to_vec();
+}
+
+#[wasm_bindgen]
+pub fn keccak_digest(s: &[u8]) -> Vec<u8> {
+  return Keccak256::digest(s).as_slice().to_vec();
+}
+```
+
+First, we use the rustwasmc tool to compile the Rust source code into WebAssembly bytecode functions using Rust 1.50 or less.
+
+```bash
+$ rustup default 1.50.0
+$ cd rust_bindgen_funcs
+$ rustwasmc build
+# The output WASM will be pkg/rust_bindgen_funcs_lib_bg.wasm
+```
+
+The [Golang source code](https://github.com/second-state/WasmEdge-go-examples/blob/master/go_BindgenFuncs/bindgen_funcs.go) to run the WebAssembly function in WasmEdge is as follows. The ExecuteBindgen() function calls the WebAssembly function and passes the call parameters using the #[wasm_bindgen] convention.
+
+
+```go
+package main
+
+import (
+    "fmt"
+    "os"
+    "github.com/second-state/WasmEdge-go/wasmedge"
+)
+
+func main() {
+    /// Expected Args[0]: program name (./bindgen_funcs)
+    /// Expected Args[1]: wasm or wasm-so file (rust_bindgen_funcs_lib_bg.wasm))
+
+    wasmedge.SetLogErrorLevel()
+
+    var conf = wasmedge.NewConfigure(wasmedge.WASI)
+    var vm = wasmedge.NewVMWithConfig(conf)
+    var wasi = vm.GetImportObject(wasmedge.WASI)
+    wasi.InitWasi(
+        os.Args[1:],     /// The args
+        os.Environ(),    /// The envs
+        []string{".:."}, /// The mapping directories
+        []string{},      /// The preopens will be empty
+    )
+
+    /// Instantiate wasm
+    vm.LoadWasmFile(os.Args[1])
+    vm.Validate()
+    vm.Instantiate()
+
+    /// Run bindgen functions
+    var res interface{}
+    var err error
+    
+    res, err = vm.ExecuteBindgen("say", wasmedge.Bindgen_return_array, []byte("bindgen funcs test"))
+    if err == nil {
+        fmt.Println("Run bindgen -- say:", string(res.([]byte)))
+    } 
+    res, err = vm.ExecuteBindgen("obfusticate", wasmedge.Bindgen_return_array, []byte("A quick brown fox jumps over the lazy dog"))
+    if err == nil {
+        fmt.Println("Run bindgen -- obfusticate:", string(res.([]byte)))
+    } 
+    res, err = vm.ExecuteBindgen("lowest_common_multiple", wasmedge.Bindgen_return_i32, int32(123), int32(2))
+    if err == nil {
+        fmt.Println("Run bindgen -- lowest_common_multiple:", res.(int32))
+    } 
+    res, err = vm.ExecuteBindgen("sha3_digest", wasmedge.Bindgen_return_array, []byte("This is an important message"))
+    if err == nil {
+        fmt.Println("Run bindgen -- sha3_digest:", res.([]byte))
+    } 
+    res, err = vm.ExecuteBindgen("keccak_digest", wasmedge.Bindgen_return_array, []byte("This is an important message"))
+    if err == nil {
+        fmt.Println("Run bindgen -- keccak_digest:", res.([]byte))
+    } 
+
+    vm.Delete()
+    conf.Delete()
+}
+```
+
+Next, let's build the Golang application with the WasmEdge Golang SDK.
+
+```bash
+$ go get -u github.com/second-state/WasmEdge-go/wasmedge
```

Comment:
It's better to add the version: `go get github.com/second-state/WasmEdge-go/wasmedge@v0.9.0-rc3`, or the `v0.8.2` will be get.